CommonApi/src/main/java/mezz/jei/api/registration/IAdvancedRegistration.java

```diff
@@ -17,4 +20,25 @@ public interface IAdvancedRegistration {
 	 * Register your own {@link IRecipeManagerPlugin} here.
 	 */
 	void addRecipeManagerPlugin(IRecipeManagerPlugin recipeManagerPlugin);
+
+	/**
+	 * Register a {@link IGlobalRecipeCategoryExtension} for a recipe category.
+	 *
+	 * @since 15.1.0
+	 */
+	<T> void addGlobalRecipeCategoryExtension(IRecipeCategory<T> recipeCategory, IGlobalRecipeCategoryExtension<T> extension);
+
+	/**
+	 * Register a {@link IGlobalRecipeCategoryExtension} for a recipe class.
+	 *
+	 * @since 15.1.0
+	 */
+	<T> void addGlobalRecipeCategoryExtension(Class<T> recipeClass, IGlobalRecipeCategoryExtension<T> extension);
```

Comment:
I want to avoid using `Class` wherever possible, the `RecipeType<T>` is a lot safer from a validation perspective. It's harder to make a mistake calling a method where the compiler wants a `RecipeType<T>`.
We should remove this method.